pkg/codegen/go/gen.go

```diff
@@ -1119,6 +1150,78 @@ func (pkg *pkgContext) genPlainType(w io.Writer, name, comment, deprecationMessa
 	fmt.Fprintf(w, "}\n\n")
 }
 
+// Returns the default value used to perform "is default" checks.
+func (pkg *pkgContext) defaultComparisonValue(t schema.Type) string {
```

Comment:
Are you looking for some variation of this? Note that the reflect runtime version is easier to unit-test out.

```
package main

import (
	"fmt"
	"reflect"
)

func isZero(v interface{}) bool {
	if v == nil {
		return true
	}
	return reflect.ValueOf(v).IsZero()
}

func check(x interface{}) {
	fmt.Printf("isZero(%v) == %v\n", x, isZero(x))
}

type foo struct {
	a int
	b *foo
}

type bar interface {
	foo()
}

func main() {
	check(1)
	check(0)
	check("")
	check("a")
	check(foo{})
	check(&foo{}) // NOTE: this is false, as the pointer is non-zero; can code a special case for this
	var b bar
	check(b)
}
```